packages/client/hmi-client/src/services/models/funman-service.ts

```diff
@@ -105,25 +106,24 @@ export const processFunman = (result: any) => {
 						return obj;
 					}, {});
 
-				const timesteps = [
-					...new Set(
-						Object.keys(filteredVals).map((key) => {
-							const splitKey = key.split('_');
-							return +splitKey[splitKey.length - 1]; // timestep
-						})
-					)
-				].sort((a, b) => a - b);
-
 				timesteps.forEach((t) => {
+					let pushFlag = true;
 					const traj: any = {
 						boxId,
 						pointId: point.id,
 						timestep: t
 					};
 					states.forEach((s) => {
-						traj[s] = filteredVals[`${s}_${t}`];
+						// Only push states that have a timestep key pair
+						if (Object.keys(filteredVals).includes(`${s}_${t}`)) {
+							traj[s] = filteredVals[`${s}_${t}`];
```

Comment:
It looks like you want to use `hasOwnProperty` or `hasOwn` instead of this "keys(...).includes(...)" pattern.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty